### PR TITLE
enable dynamic image building via dynamic_image_building profile option

### PR DESCRIPTION
**Pull request by batpad**
For now this, just comments out the "Build your Own Image" functionality.

@yuvipanda I think this is what we want for now. This removes "Build your Own Image" for now. I think eventually we need to figure out exactly how we want this defined in the profile config, and then it should be trivial to conditionally add it when configured in the profile options.

cc @oliverroick

### CODE REVIEW

**Review by yuvipanda — CHANGES_REQUESTED** [review]
@batpad unfortunately there are existing hubs using this functionality, so we have to keep it.

**Comment by yuvipanda** [comment]
Had a nice quick phone conversation with @batpad. 

https://github.com/2i2c-org/infrastructure/blob/9a2e739fd82445e02cd91e21c4b5b48e40ff1810/config/clusters/2i2c/imagebuilding-demo.values.yaml#L36 is an exsiting hub using this config.

I would suggest that we look for a key called `dynamic_image_building` (at the same level as `unlisted_choice`. If `dynamic_image_building.enabled` is set to `True`, *and* `unlisted_choice.enabled` is set to `True`, we display the ImageBuilder. This allows us to specify further config (such as location of the binderhub API) in the future under `dynamic_image_building`

**Comment by batpad** [comment]
Soo .. I started working on this with this commit: https://github.com/yuvipanda/jupyterhub-fancy-profiles/pull/58/commits/3aa57d796e3feae9f05bbd822de137eb4c936399

This mostly worked, except for a bug when: if `unlisted_choice` and `dynamic_image_building` are both enabled, but there are no `choices` specified, then it does not show the options drop-down at all, whereas in this case it should show the drop-down and have Bring Your Own Image and Dynamic Image Building as the two options in the drop-down. This is because we assume that if choices is empty, there can only be one option here: https://github.com/yuvipanda/jupyterhub-fancy-profiles/blob/main/src/hooks/useSelectOptions.js#L5

So to finish up this PR, we need to:

 - [ ] Fix the above mentioned bug and calculate `hasDefaultOptions` correctly if both unlisted_choice and dynamic_image_building are enabled.
 - [ ] Write tests for this behaviour

I think I can work on this later in the week.

**Comment by batpad** [comment]
@oliverroick sorry, I just can't get the hang of how selectors work in these tests .. I added a test config to the setupTests.py but then have some commented out code where the test should be .. would you be able to help / take a look?

File: src/ProfileForm.test.js [hunk 1]
```diff
@@ -186,3 +186,12 @@ test("profile marked as default is selected by default", () => {
...
   expect(nonDefaultRadio.checked).toBeFalsy();
 });
+
+test("having dynamic_image_building enabled and no other choices shows dropdown", () => {
+  // const { container } = render(
+  //   <SpawnerFormProvider>
+  //     <ProfileForm />
+  //   </SpawnerFormProvider>,
+  // );
+  // check that the dropdown select exists with two choices
```
Comment (batpad) [inline src/ProfileForm.test.js:196]: @oliverroick this is what I failed to figure out how to do ..

**Pull request merged**